enhancements/machine-config/enable-evented-pleg-in-openshift.md

```diff
@@ -0,0 +1,196 @@
+  - Users can disable this feature to make kubelet use existing relisting based PLEG.
+- Another risk is the CRI implementation could have a buggy event emitting system, and miss pod lifecycle events.
+  - A mitigation is a `kube_pod_missed_events` metric, which the Kubelet could report when a lifecycle event is registered that wasn't triggered by an event, but rather by changes of state between lists.
+  - While using the Evented implementation, the periodic relisting functionality would still be used with an increased interval which should work as a fallback mechanism for missed events in case of any disruptions.
+  - Evented PLEG will need to update global cache timestamp periodically in order to make sure pod workers don't get stuck at [GetNewerThan](https://github.com/kubernetes/kubernetes/blob/4a894be926adfe51fd8654dcceef4ece89a4259f/pkg/kubelet/pod_workers.go#L924) in case Evented PLEG misses the event for any unforeseen reason.
+
+### Drawbacks
+
+This feature requires changes in the code base of the PLEG module that adjusts the container runtime state, accordngly updates the pod cache.
+Kubelet depends on the events generated by the container runtimes(Evented PLEG) to update the pod cache.
+Sometimes, there is a chance of events getting missed and hence relying alone on the evented pleg, without having sufficient benchmark data, is not recommended.
+
+One more minor drawback of this feature is that the node gets updated/rebooted twice with the two newly rendered machine configs
+i.e. one due to the Kubelet Config Controller and the second due to the Container Runtime Config Controller of the MCO.
+
+## Design Details
+
+### Test Plan
+- Unit tests must be added to the MCO that monitors the newly added field, updates the kubelet, cri-o configurations.
+- e2e bootstrap related testcases must be added to compare the newly rendered, bootstrapped machine config objects.
+- An openshift CI job must be added to test the node e2e tests by enabling the evented pleg feature.
+
+### Graduation Criteria
+#### Dev Preview -> Tech Preview
+`Evented PLEG` will be a tech preview feature on its initial release as this is a featuregated, beta feature in upstream.
+There must be a sufficient internal testing performed along with the gathering of the customer feedback before graduating this feature to GA.
+#### Tech Preview -> GA
+Graduation requirements to GA are:
+
+* No regressions from the Generic PLEG to Evented PLEG
+* Processes are correctly OOMed
+* No performance issues - PSAP and QE teams will be asked to test their suites for regressions
+* Metrics are accurate and correctly submitted to monitoring
+* Upstream feature graduation to GA
+* Add blocking upgrade ci jobs with the evented pleg feature enabled
+* OpenShift's evented pleg enabled upgrade jobs pass percentage is similar or better than the OpenShift's Generic PLEG(default) upgrade job pass percentage
+
+#### Removing a deprecated feature
+N/A
+### Upgrade / Downgrade Strategy
```

Comment:
As per my understanding, In OCP, enabling tech-preview features leads to cluster not being upgradable. If I understand correctly points noted here are relevant only when feature goes GA in upstream?